torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -1991,10 +2015,608 @@ RegisterOperators reg2({
           return 0;
         }),
 
+    Operator(
```

Comment:
There is no need to manually write stack manipulation code for these functions, and maintaining that code as we change the interface will become a burden.

Look at https://github.com/pytorch/pytorch/blob/2de1020b3ed6f8fadc102f68f3b11bcab93db8df/torch/csrc/jit/register_prim_ops.cpp#L3187 for examples of how it should be written.

Reply:
this is a brutal API for debugging...

how would I return a tuple? should it just work? I have no idea which of the 25 functions is broken right now after a mass conversion :(

```
In file included from ../aten/src/ATen/core/op_registration/op_registration.h:9:0,
                 from ../torch/csrc/jit/custom_operator.h:5,
                 from ../torch/csrc/jit/register_string_ops.cpp:2:
../aten/src/ATen/core/op_registration/kernel_functor.h: In instantiation of ‘struct c10::detail::return_type_to_ivalue<int, void>’:
../aten/src/ATen/core/op_registration/kernel_functor.h:216:44:   required from ‘static c10::IValue c10::detail::legacy_return_type_to_ivalue<T, Enable>::call(T_&&) [with T_ = int; T = $
nt; Enable = void]’
../aten/src/ATen/core/op_registration/kernel_functor.h:244:88:   required from ‘static void c10::detail::push_outputs<OutputType, AllowDeprecatedTypes>::call(OutputType&&, c10::Stack*)
[with OutputType = int; bool AllowDeprecatedTypes = true; c10::Stack = std::vector<c10::IValue>]’
../aten/src/ATen/core/op_registration/kernel_functor.h:272:115:   required from ‘static void c10::detail::wrap_kernel_functor<KernelFunctor, AllowDeprecatedTypes, typename std::enable_$
f<(! std::is_same<void, typename c10::guts::infer_function_traits<Functor>::type::return_type>::value), void>::type>::call(c10::Stack*, c10::KernelCache*) [with KernelFunctor = c10::de$
ail::WrapRuntimeKernelFunctor_<torch::jit::{anonymous}::<lambda(std::__cxx11::string, std::__cxx11::string, int64_t, int64_t)>, int, c10::guts::typelist::typelist<std::__cxx11::basic_s$
ring<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, long int, long int> >; bool AllowDeprecatedT$
pes = true; c10::Stack = std::vector<c10::IValue>]’
../aten/src/ATen/core/op_registration/op_registration.h:270:9:   required from ‘c10::RegisterOperators::Options&& c10::RegisterOperators::Options::kernelFunctor(c10::optional<c10::Tens$
rTypeId>&&, ConstructorParameters&& ...) && [with KernelFunctor = c10::detail::WrapRuntimeKernelFunctor_<torch::jit::{anonymous}::<lambda(std::__cxx11::string, std::__cxx11::string, in$
64_t, int64_t)>, int, c10::guts::typelist::typelist<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<c$
ar>, std::allocator<char> >, long int, long int> >; bool AllowDeprecatedTypes = true; ConstructorParameters = {torch::jit::{anonymous}::<lambda(std::__cxx11::string, std::__cxx11::stri$
g, int64_t, int64_t)>}]’
../aten/src/ATen/core/op_registration/op_registration.h:391:201:   required from ‘c10::guts::enable_if_t<c10::guts::is_functor<LambdaType>::value, c10::RegisterOperators&&> c10::Regist$
rOperators::op(const string&, FuncType&&, c10::RegisterOperators::Options&&) && [with FuncType = torch::jit::{anonymous}::<lambda(std::__cxx11::string, std::__cxx11::string, int64_t, i$
t64_t)>; c10::guts::enable_if_t<c10::guts::is_functor<LambdaType>::value, c10::RegisterOperators&&> = c10::RegisterOperators&&; std::__cxx11::string = std::__cxx11::basic_string<char>]$
../aten/src/ATen/core/op_registration/op_registration.h:325:5:   [ skipping 5 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/home/bwasti/bin/gcc-7.3.0/include/c++/7.3.0/bits/shared_ptr_base.h:1295:35:   required from ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_make_shared_tag, const _Alloc&, _Args&&
...) [with _Alloc = std::allocator<c10::RegisterOperators>; _Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, torch::jit::{anonymous}::<l$
mbda(std::__cxx11::string, std::__cxx11::string, int64_t, int64_t)>}; _Tp = c10::RegisterOperators; __gnu_cxx::_Lock_policy _Lp = (__gnu_cxx::_Lock_policy)2]’
/home/bwasti/bin/gcc-7.3.0/include/c++/7.3.0/bits/shared_ptr.h:344:64:   required from ‘std::shared_ptr<_Tp>::shared_ptr(std::_Sp_make_shared_tag, const _Alloc&, _Args&& ...) [with _Al$
oc = std::allocator<c10::RegisterOperators>; _Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, torch::jit::{anonymous}::<lambda(std::__cx$
11::string, std::__cxx11::string, int64_t, int64_t)>}; _Tp = c10::RegisterOperators]’
/home/bwasti/bin/gcc-7.3.0/include/c++/7.3.0/bits/shared_ptr.h:690:14:   required from ‘std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = c10::RegisterO$
erators; _Alloc = std::allocator<c10::RegisterOperators>; _Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, torch::jit::{anonymous}::<lam$
da(std::__cxx11::string, std::__cxx11::string, int64_t, int64_t)>}]’
/home/bwasti/bin/gcc-7.3.0/include/c++/7.3.0/bits/shared_ptr.h:706:39:   required from ‘std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [with _Tp = c10::RegisterOperators; _Args = {$
onst std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, torch::jit::{anonymous}::<lambda(std::__cxx11::string, std::__cxx11::string, int64_t, int64_t)>}]$
../torch/csrc/jit/custom_operator.h:260:70:   required from ‘torch::jit::RegisterOperators& torch::jit::RegisterOperators::op(const string&, Implementation&&) [with Implementation = to$
ch::jit::{anonymous}::<lambda(std::__cxx11::string, std::__cxx11::string, int64_t, int64_t)>; std::__cxx11::string = std::__cxx11::basic_string<char>]’
../torch/csrc/jit/register_string_ops.cpp:260:10:   required from here
../aten/src/ATen/core/op_registration/kernel_functor.h:208:5: error: static assertion failed: You tried to register a kernel with an unsupported integral return argument type. Please u$
e int64_t instead.
     static_assert(guts::false_t<T>::value, "You tried to register a kernel with an unsupported integral return argument type. Please use int64_t instead.");
```